.github/deployment/scripts/protocol-cli.rs

```diff
@@ -801,6 +890,166 @@ async fn update_chain_config(config: &DeploymentConfig, addresses_file: &str, ch
     std::process::exit(0);
 }
 
+// =============================================================================
+// VERIFY CONTRACTS
+// =============================================================================
+
+struct NetworkInfo {
+    chain_id: u64,
+    network: &'static str,
+}
+
+/// Gets network information including chain ID and environment (testnet/mainnet)
+fn get_network_info(network: &str) -> Option<NetworkInfo> {
+    Some(match network {
+        "base-sepolia" => NetworkInfo { chain_id: 84532, network: "testnet" },
+        "base" => NetworkInfo { chain_id: 8453, network: "mainnet" },
+        "optimism-sepolia" => NetworkInfo { chain_id: 11155420, network: "testnet" },
+        "optimism" => NetworkInfo { chain_id: 10, network: "mainnet" },
+        "arbitrum-sepolia" => NetworkInfo { chain_id: 421614, network: "testnet" },
+        "arbitrum" => NetworkInfo { chain_id: 42161, network: "mainnet" },
+        "ethereum-sepolia" => NetworkInfo { chain_id: 11155111, network: "testnet" },
+        "ethereum-mainnet" => NetworkInfo { chain_id: 1, network: "mainnet" },
+        "pecorino-signet" => NetworkInfo { chain_id: 14174, network: "testnet" },
+        "pecorino-host" => NetworkInfo { chain_id: 3151908, network: "testnet" },
+        _ => return None,
+    })
+}
+
+/// Encodes constructor arguments based on contract type
+fn encode_constructor_args(
+    constructor_type: &ConstructorType,
+    contract_name: &str,
+    env_vars: &HashMap<String, String>,
+) -> Result<String> {
+    match constructor_type {
+        ConstructorType::None | ConstructorType::Core => Ok(String::new()),
+        ConstructorType::Owner => {
+            let owner = env_vars.get("OWNER_ADDRESS")
+                .ok_or_else(|| anyhow!("OWNER_ADDRESS not found in environment"))?;
+            Ok(format!("000000000000000000000000{}", owner.trim_start_matches("0x")))
```

Comment:
nit: since the padding is always the same - I'd at least make `000000000000000000000000` a constant to avoid human error.